File: dotnet/src/CarespaceSDK/Api/UsersApi.cs
using CarespaceSDK.Http;
using CarespaceSDK.Models;

namespace CarespaceSDK.Api;

public class UsersApi : IUsersApi
{
    private readonly IHttpClient _httpClient;

    public UsersApi(IHttpClient httpClient)
    {
        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
    }

    public async Task<PaginatedResponse<User>> GetUsersAsync(
        int page = 1, 
        int limit = 20, 
        string? search = null, 
        UserRole? role = null, 
        bool? isActive = null,
        CancellationToken cancellationToken = default)
    {
        var queryParams = new List<string>();
        queryParams.Add($"page={page}");
        queryParams.Add($"limit={limit}");
        
        if (!string.IsNullOrEmpty(search))
            queryParams.Add($"search={Uri.EscapeDataString(search)}");
        
        if (role.HasValue)
            queryParams.Add($"role={role.Value}");
        
        if (isActive.HasValue)
            queryParams.Add($"is_active={isActive.Value}");

        var query = string.Join("&", queryParams);
        var endpoint = $"/users?{query}";

        return await _httpClient.GetAsync<PaginatedResponse<User>>(endpoint, cancellationToken);
    }

    public async Task<ApiResponse<User>> GetUserAsync(string userId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentException("User ID is required", nameof(userId));
        return await _httpClient.GetAsync<ApiResponse<User>>($"/users/{userId}", cancellationToken);
    }

    public async Task<ApiResponse<User>> GetUserProfileAsync(CancellationToken cancellationToken = default)
    {
        return await _httpClient.GetAsync<ApiResponse<User>>("/users/profile", cancellationToken);
    }

    public async Task<ApiResponse<User>> CreateUserAsync(CreateUserRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<User>>("/users", request, cancellationToken);
    }

    public async Task<ApiResponse<User>> UpdateUserAsync(string userId, UpdateUserRequest request, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentException("User ID is required", nameof(userId));
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PutAsync<ApiResponse<User>>($"/users/{userId}", request, cancellationToken);
    }

    public async Task<ApiResponse<User>> UpdateUserProfileAsync(UpdateUserRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PutAsync<ApiResponse<User>>("/users/profile", request, cancellationToken);
    }

    public async Task DeleteUserAsync(string userId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentException("User ID is required", nameof(userId));
        await _httpClient.DeleteAsync($"/users/{userId}", cancellationToken);
    }

    public async Task<ApiResponse<object>> DeactivateUserAsync(string userId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentException("User ID is required", nameof(userId));
        return await _httpClient.PostAsync<ApiResponse<object>>($"/users/{userId}/deactivate", null, cancellationToken);
    }

    public async Task<ApiResponse<object>> ActivateUserAsync(string userId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentException("User ID is required", nameof(userId));
        return await _httpClient.PostAsync<ApiResponse<object>>($"/users/{userId}/activate", null, cancellationToken);
    }
}

File: temp-repos/carespace-kotlin-sdk/carespace-sdk/src/commonMain/kotlin/com/carespace/sdk/api/UsersApi.kt
package com.carespace.sdk.api

import com.carespace.sdk.http.CarespaceHttpClient
import com.carespace.sdk.models.*

/**
 * Users API interface
 */
interface UsersApi {
    suspend fun getUsers(
        page: Int = 1,
        limit: Int = 20,
        search: String? = null,
        role: UserRole? = null,
        isActive: Boolean? = null
    ): PaginatedResponse<User>

    suspend fun getUser(userId: String): ApiResponse<User>
    suspend fun getUserProfile(): ApiResponse<User>
    suspend fun createUser(request: CreateUserRequest): ApiResponse<User>
    suspend fun updateUser(userId: String, request: UpdateUserRequest): ApiResponse<User>
    suspend fun updateUserProfile(request: UpdateUserRequest): ApiResponse<User>
    suspend fun deleteUser(userId: String)
    suspend fun deactivateUser(userId: String): ApiResponse<Unit>
    suspend fun activateUser(userId: String): ApiResponse<Unit>
}

/**
 * Users API implementation
 */
class UsersApiImpl(private val httpClient: CarespaceHttpClient) : UsersApi {

    override suspend fun getUsers(
        page: Int,
        limit: Int,
        search: String?,
        role: UserRole?,
        isActive: Boolean?
    ): PaginatedResponse<User> {
        val parameters = buildMap {
            put("page", page)
            put("limit", limit)
            search?.let { put("search", it) }
            role?.let { put("role", it.name.lowercase()) }
            isActive?.let { put("is_active", it) }
        }
        return httpClient.get("/users", parameters)
    }

    override suspend fun getUser(userId: String): ApiResponse<User> {
        require(userId.isNotBlank()) { "User ID cannot be blank" }
        return httpClient.get("/users/$userId")
    }

    override suspend fun getUserProfile(): ApiResponse<User> {
        return httpClient.get("/users/profile")
    }

    override suspend fun createUser(request: CreateUserRequest): ApiResponse<User> {
        return httpClient.post("/users", request)
    }

    override suspend fun updateUser(userId: String, request: UpdateUserRequest): ApiResponse<User> {
        require(userId.isNotBlank()) { "User ID cannot be blank" }
        return httpClient.put("/users/$userId", request)
    }

    override suspend fun updateUserProfile(request: UpdateUserRequest): ApiResponse<User> {
        return httpClient.put("/users/profile", request)
    }

    override suspend fun deleteUser(userId: String) {
        require(userId.isNotBlank()) { "User ID cannot be blank" }
        httpClient.delete("/users/$userId")
    }

    override suspend fun deactivateUser(userId: String): ApiResponse<Unit> {
        require(userId.isNotBlank()) { "User ID cannot be blank" }
        return httpClient.post("/users/$userId/deactivate")
    }

    override suspend fun activateUser(userId: String): ApiResponse<Unit> {
        require(userId.isNotBlank()) { "User ID cannot be blank" }
        return httpClient.post("/users/$userId/activate")
    }
}